pkg/storage/store.go

```diff
@@ -16,14 +16,17 @@ var ErrValueNotFound = errors.New("store does not have a value associated with t
 
 // Provider represents a storage provider.
 type Provider interface {
-	// OpenStore opens a store with the given name and returns it.
+	// OpenStore opens a store with the given name and returns it. If no store exists under the given name, it is created.
 	OpenStore(name string) (Store, error)
 
 	// CloseStore closes the store with the given name.
 	CloseStore(name string) error
 
 	// Close closes all stores created under this store provider.
 	Close() error
+
+	// StoreExists checks whether a store exists under the given name already.
+	StoreExists(name string) (bool, error)
```

Comment:
@fqutishat If you recall, I had an issue in the EDV repo where I needed the ability to know if a store is open without actually opening it. The solution we had before was to track the open stores in a map in the operations.go file. However, I realized while working on this story that that strategy wouldn't work for a persistent store, since now there can be stores already created from running this executable previously. Thus, I need to ask the provider if a store is open first. So I added this "StoreExists" method to the Provider interface. Let me know if there's another way to do this.

Reply:
you don't need it part of interface

Reply:
Per our discussion, I added the CreateStore method and made the OpenStore method only open existing stores.